### PR TITLE
[radiocanada] Extractor Enhancements

### PR DESCRIPTION
**Pull request by up4**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x ] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x ] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

I generalized video id extraction for all pages, except the home page, on the [Radio-Canada website](http://ici.radio-canada.ca), the french language federal public broadcaster.

### CODE REVIEW

File: youtube_dl/extractor/radiocanada.py [hunk 1]
```diff
@@ -168,3 +169,30 @@ class RadioCanadaAudioVideoIE(InfoExtractor):
 
     def _real_extract(self, url):
         return self.url_result('radiocanada:medianet:%s' % self._match_id(url))
+
+
+
+class RadioCanadaArticleIE(InfoExtractor):
+    'radiocanada:article'
```
Comment (dstftw) [inline youtube_dl/extractor/radiocanada.py]: What's that string literal for?

File: youtube_dl/extractor/radiocanada.py [hunk 2]
```diff
@@ -168,3 +169,30 @@ class RadioCanadaAudioVideoIE(InfoExtractor):
...
+
+        webpage = unescapeHTML(self._download_webpage(url, display_id))
+
+
+        entries = [
+            self.url_result(
+                'radiocanada:medianet:%s' % mobj.group('id'),
+                ie=RadioCanadaIE.ie_key(), video_id=mobj.group('id'))
+            for mobj in re.finditer(
+                r'\"idMedia\"\s*:\s*\"(?P<id>\d+)\"', webpage)]
```
Comment (dstftw) [inline youtube_dl/extractor/radiocanada.py]: No escape for `"` is necessary.

File: youtube_dl/extractor/radiocanada.py [hunk 3]
```diff
@@ -168,3 +169,30 @@ class RadioCanadaAudioVideoIE(InfoExtractor):
 
     def _real_extract(self, url):
         return self.url_result('radiocanada:medianet:%s' % self._match_id(url))
+
+
+
+class RadioCanadaArticleIE(InfoExtractor):
+    'radiocanada:article'
+    _VALID_URL = r'https?://ici\.radio-canada\.ca/(?P<id>[^?#&]+)'
+
```
Comment (dstftw) [inline youtube_dl/extractor/radiocanada.py]: Add tests.

File: youtube_dl/extractor/radiocanada.py [hunk 4]
```diff
@@ -168,3 +169,30 @@ class RadioCanadaAudioVideoIE(InfoExtractor):
 
     def _real_extract(self, url):
         return self.url_result('radiocanada:medianet:%s' % self._match_id(url))
+
+
+
```
Comment (dstftw) [inline youtube_dl/extractor/radiocanada.py]: flake8.

**Comment by dstftw** [comment]
```
PS C:\dev\youtube-dl\master> py -2.6 .\test\test_all_urls.py
...F.............
======================================================================
FAIL: test_no_duplicates (__main__.TestAllURLsMatching)
----------------------------------------------------------------------
Traceback (most recent call last):
  File ".\test\test_all_urls.py", line 98, in test_no_duplicates
    '%s should not match URL %r . That URL belongs to %s.' % (type(ie).__name__, url, tc['name']))
AssertionError: RadioCanadaArticleIE should not match URL u'http://ici.radio-canada.ca/widgets/mediaconsole/medianet/7184272' . That URL belongs to RadioCanada.

----------------------------------------------------------------------
Ran 17 tests in 8.683s

FAILED (failures=1)
```

File: youtube_dl/extractor/radiocanada.py (hunk 2 above)
Comment (up4) [inline youtube_dl/extractor/radiocanada.py]: will remove

File: youtube_dl/extractor/radiocanada.py (hunk 3 above)
Comment (up4) [inline youtube_dl/extractor/radiocanada.py]: will do

**Comment by up4** [comment]
Did changes for everything. the URL test doesn't pass because it does not consider the `suitable` class method that discriminate against duplicates and consider the regex for `RadioCanadaArticleIE` to be a duplicate of `RadioCanadaIE`. How should I handle this?

**Comment by timendum** [comment]
Can you restrict the _VALID_URL of RadioCanadaArticleIE or RadioCanadaIE to check if an url belongs to one extractor or the other?

**Pull request closed without merging**